mlserver/kafka/server.py

```diff
@@ -0,0 +1,110 @@
+from mlserver.errors import MLServerError
+from typing import Dict
+import orjson
+from enum import Enum
+from aiokafka import AIOKafkaConsumer, AIOKafkaProducer
+
+from ..cloudevents import get_cloudevent_headers
+from ..handlers import DataPlane, ModelRepositoryHandlers
+from ..settings import Settings
+from ..types import InferenceRequest
+from ..model import MLModel
+from ..utils import logger
+
+
+class KafkaServer:
+    def __init__(
+        self,
+        settings: Settings,
+        data_plane: DataPlane,
+        model_repository_handlers: ModelRepositoryHandlers,
+    ):
+        self._settings = settings
+        self._data_plane = data_plane
+        self._model_repository_handlers = model_repository_handlers
+
+    def _create_server(self):
+        self._consumer = AIOKafkaConsumer(
+            self._settings.kafka_topic_input,
+            bootstrap_servers=self._settings.kafka_servers,
+        )
+        self._producer = AIOKafkaProducer(
+            bootstrap_servers=self._settings.kafka_servers
+        )
+
+    async def add_custom_handlers(self, model: MLModel):
+        # TODO: Implement
+        pass
+
+    async def delete_custom_handlers(self, model: MLModel):
+        # TODO: Implement
+        pass
+
+    async def start(self):
+        self._create_server()
+
+        # TODO: Move logic to KafkaRouter (Also to implement cutom handlers modularly)
+        await self._consumer.start()
+        await self._producer.start()
+        # TODO: Catch or ensure parent function catches and stops server
```

Comment:
This is an interesting point. What's the recommended way to deal with errors with Kafka? In REST / gRPC is easy as they're sync protocols, but is there any standard practice for Kafka?